service/resource/project/resource.go

```diff
@@ -243,25 +249,30 @@ func (r *Resource) ProcessUpdateState(obj, updateState interface{}) error {
 		r.logger.Log("debug", "updating projects in the Kubernetes cluster")
 
 		for _, p := range projectsToUpdate {
+			ID, err := strconv.Atoi(p.ID)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
 			instErr := r.installer.Install(installerspec.Project{Name: p.Name, Ref: p.Ref})
 			if instErr != nil {
-				evenErr := r.eventer.SetFailedStatus(eventerspec.DeploymentEvent{ID: 0, Name: p.Name, Sha: p.Ref})
+				evenErr := r.eventer.SetFailedStatus(eventerspec.DeploymentEvent{ID: ID, Name: p.Name, Sha: p.Ref})
```

Comment:
I am not happy with this part here because it is a massive copy/paste pile. I am not improving this now. #later